src/io/local_storage.ts

```diff
@@ -0,0 +1,227 @@
+/**
+ * @license
+ * Copyright 2018 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+// tslint:disable:max-line-length
+import {arrayBufferToBase64String, base64StringToArrayBuffer, stringByteLength} from './io_utils';
+import {IOHandler, ModelArtifacts, ModelArtifactsInfo, SaveResult} from './types';
+
+// tslint:enable:max-line-length
+
+const PATH_SEPARATOR = '/';
+const PATH_PREFIX = 'tensorflowjs_models';
+const INFO_SUFFIX = 'info';
+const MODEL_TOPOLOGY_SUFFIX = 'model_topology';
+const WEIGHT_SPECS_SUFFIX = 'weight_specs';
+const WEIGHT_DATA_SUFFIX = 'weight_data';
+
+/**
+ * Purge all tensorflow.js-saved model artifacts from local storage.
+ *
+ * @returns Paths of the models purged.
+ */
+export function purgeLocalStorageArtifacts(): string[] {
+  if (!(window && window.localStorage)) {
+    throw new Error(
+        'purgeLocalStorageModels() cannot proceed because local storage is ' +
+        'unavailable in the current environment.');
+  }
+  const LS = window.localStorage;
+  const purgedModelPaths: string[] = [];
+  for (let i = 0; i < LS.length; ++i) {
+    const key = LS.key(i);
+    const prefix = PATH_PREFIX + PATH_SEPARATOR;
+    if (key.startsWith(prefix) && key.length > prefix.length) {
+      const modelName = key.slice(prefix.length).split(PATH_PREFIX)[0];
+      if (purgedModelPaths.indexOf(modelName) === -1) {
+        purgedModelPaths.push(modelName);
+      }
+    }
+  }
+  return purgedModelPaths;
+}
+
+/**
+ * IOHandler subclass: Browser Local Storage.
+ *
+ * See the doc string to `browserLocalStorage` for more details.
+ */
+export class BrowserLocalStorage implements IOHandler {
+  protected readonly LS: Storage;
+  protected readonly modelPath: string;
+  protected readonly paths: {[key: string]: string};
+
+  constructor(modelPath: string) {
+    if (!(window && window.localStorage)) {
+      throw new Error(
+          'The current environment does not support local storage.');
+    }
+    this.LS = window.localStorage;
+
+    if (modelPath == null) {
+      throw new Error(
+          'For local storage, modelPath must not be null or undefined.');
```

Comment:
I would suggest using the same error string for both this Error and the next one.  We don't people playing whack-a-mole.  Following the discussion last year on TF errors, something like "Local Storage requires a modelPath that is no empty, underfined, nor null."  Maybe add on a "Suggested fix/use: use a unique descriptive name, possibly including a version token such as 'supermodel-01'"